src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp

```diff
@@ -68,7 +69,7 @@ struct ONNXLoopOpLowering : public ConversionPattern {
     {
       OpBuilder::InsertionGuard insertGuard(rewriter);
 
-      auto condReg = rewriter.create<KrnlLoadOp>(loc, cond).getResult();
+      Value condReg = createKrnl.load(cond);
```

Comment:
Interesting, the builders were intended to work with the new interface, and here the loop code is still manually changing insertion points. But this works as onnx-mlir builders only keep a reference to the original `rewriter` as shown below
```c++
struct DialectBuilder {
  DialectBuilder(OpBuilder &b, Location loc) : b(b), loc(loc) {}
  DialectBuilder(DialectBuilder &db) : b(db.b), loc(db.loc) {}
  virtual ~DialectBuilder() {}
  DialectBuilder(DialectBuilder &&) = delete;
  DialectBuilder &operator=(const DialectBuilder &) = delete;
  DialectBuilder &&operator=(const DialectBuilder &&) = delete;

  OpBuilder &getBuilder() const { return b; }
  Location getLoc() const { return loc; }

protected:
  OpBuilder &b;
  Location loc;
};
```